coverage/xmlreport.py

```diff
@@ -182,7 +182,7 @@ def xml_file(self, fr: FileReporter, analysis: Analysis, has_arcs: bool) -> None
                 rel_name = filename[len(source_path)+1:]
                 break
         else:
-            rel_name = fr.relative_filename()
```

Comment:
If the for loop is run, the `rel_name` is generated from `filename` which is already converted from backslash to slash in line 177

When the else branch is run, `rel_name` is generated from the `FileReporter` which needs the conversion